streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImplJoin.java

```diff
@@ -69,6 +69,10 @@ public long get() {
         }
     }
 
+    static class MinTime {
```

Comment:
Could we merge this and MaxObservedStreamTime into a single class to be shared among operator nodes?